KomaMRIBase/test/runtests.jl

```diff
@@ -647,6 +646,38 @@ end
         @test yt[: ,end] ≈ rot_y .+ vy*t[end]
         @test zt[: ,end] ≈ rot_z .+ vz*t[end]
     end
+    @testset "Key Time Points" begin
+        # Sequence duration
+        t_start = 0.0
+        t_end   = 1.5 
+        # TimeCurve parameters
+        t        = [0.0, 0.1, 0.3]
+        t_unit   = [0.0, 0.4, 1.0]
+        periods  = [1.0, 0.5, 2.0]
+        periodic = true
+        dx = dy = [0.0 0.0 0.0 0.0]
+        dz      = [3.0 4.0 -4. -3.]
+        reset   = [false false true false]
+        ϵ = KomaMRIBase.MIN_RISE_TIME
+        # Key time points ("manually" determined):
+        period_times = [t+δ for t in (0.0, 0.3, 0.45, 1.05, 1.35, 1.5) for δ in (-ϵ, ϵ) if (t+δ) > t_start && (t+δ) < t_end]
+        reset_times  = [0.2, 0.4, 0.85, 1.25, 1.45] .- ϵ
+        # Any motion with no spin resets (only key time points derived from periods):
+        pth  = path(dx, dy, dz, TimeCurve(t, t_unit, periodic, periods), AllSpins())
+        seqd_t = [t_start, t_end]
+        KomaMRIBase.add_key_time_points!(seqd_t, pth)
+        @test sort(seqd_t[3:end]) ≈ period_times
```

Comment:
This is much better, but why do you need to start comparing from 3? 

I think it is better to add the missing points at the right had side to make it more explicit `[-ϵ; 0; period_times]`, or something like that.